torchinductor/lowering.py

```diff
@@ -2849,6 +2849,7 @@ def pow_native(a, b):
 
 @register_lowering(aten.pow, broadcast=True)
 def pow(a, b):
+    assert a.get_dtype() in (torch.float16, torch.float32, torch.float64), "Pow input must be floating point."
```

Comment:
Is AssertionError the right exception to throw?

Reply:
In the references we use `check` util, it can be imported here. Usually asserts are for internal things that go wrong, other errors for invalid user data. Also, why are we limiting the base to fp types? Eager accepts integer bases and type promotes the result:
```
In [9]: a=torch.randint(2, (4,))

In [10]: a**2.5
Out[10]: tensor([0., 1., 1., 0.])
```

Reply:
If we accept int types, we get a:

```
ValueError('input arg type does not match.Expect one of dict_keys([(triton.language.fp32, triton.language.fp32), (triton.language.fp64, triton.language.fp64)]), got (triton.language.int32, triton.language.int32)')
```

In triton. Are you suggesting we take the ints and turn them into floats here?

Reply:
> Is AssertionError the right exception to throw?

The AssertionError becomes a LoweringException `torchinductor.exc.LoweringException: AssertionError: Pow input must be floating point.` - But open to any others you think fit better.

Reply:
Discussed offline - type promotion is always upstream of dropping into the op, so checks in ops are a good thing. That being said, we should look at why type promotion is failing here. @ngimel will look at pow tomorrow to see what we can do around promotion rules here.

Reply:
> If we accept int types, we get a:
> 
> ```
> ValueError('input arg type does not match.Expect one of dict_keys([(triton.language.fp32, triton.language.fp32), (triton.language.fp64, triton.language.fp64)]), got (triton.language.int32, triton.language.int32)')
> ```
> 
> In triton. Are you suggesting we take the ints and turn them into floats here?

If this fails in triton but passes in cpu, then we should fix it in the triton codegen.

Reply:
I think we should (fix) type promote, but maybe we can file an issue on them also.

Reply:
RM for now.